docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,444 @@
+>
+> - 2.0.0: report a deprecation warning for accessing private operator functions in public inline functions
+> - 2.1.0: raise the warning to an error
+
+### Prohibit passing invalid arguments to invariant parameters annotated with @UnsafeVariance
+
+> **Issue**: [KTLC-72](https://youtrack.jetbrains.com/issue/KTLC-72)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now ignores `@UnsafeVariance` annotations during type checks, enforcing stricter type safety for invariant type parameters.
+> This prevents invalid calls that rely on `@UnsafeVariance` to bypass expected type checks.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: activate the new behavior
+
+### Report nullability errors for error-level nullable arguments of warning-level Java types
+
+> **Issue**: [KTLC-100](https://youtrack.jetbrains.com/issue/KTLC-100)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now detects nullability mismatches in Java methods where a warning-level
+> nullable type contains type arguments with stricter, error-level nullability.
+> This ensures that previously ignored errors in type arguments are reported correctly.
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report a deprecation warning for nullability mismatches in Java methods with stricter type arguments
+> - 2.1.0: raise the warning to an error
+
+### Report implicit usages of inaccessible types
+
+> **Issue**: [KTLC-3](https://youtrack.jetbrains.com/issue/KTLC-3)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now reports usages of inaccessible types in function literals and type arguments,
+> preventing compilation and runtime failures caused by incomplete type information.
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report warnings for function literals with parameters or receivers of inaccessible non-generic types and for types with inaccessible type argument; report errors for
+> function literals with parameters or receivers of inaccessible generic types and for types with inaccessible generic type arguments in specific scenarios
+> - 2.1.0: raise warnings to errors for function literals with parameters and receivers of inaccessible non-generic types
+> - 2.2.0: raise warnings to errors for types with inaccessible type arguments
+
+## Standard library
+
+### Remove kotlin-stdlib-common artifacts in compiler tests
+
+> **Issue**: [KT-62159](https://youtrack.jetbrains.com/issue/KT-62159)
+>
+> **Component**: kotlin-stdlib
+>
+> **Incompatible change type**: binary
+>
+> **Short summary**: The JAR file published under `org.jetbrains.kotlin:kotlin-stdlib-common` is deprecated, with `.klib` files replacing it as the standard format.
+> The legacy `kotlin-stdlib-common` artifacts built with `kotlin-platform-common` and `enableCompatibilityMetadataVariant` are removed in compiler tests.
```

Comment:
Hm, I'm not sure that our compiler tests are relevant to the users.

Reply:
To summarize the change: we're removing compatibility JAR with common multiplatform declarations metadata from stdlib artifacts `kotlin-stdlib` and `kotlin-stdlib-common`.

Reply:
Thank you - I removed the part about the compiler tests.
I pushed the new version like:

> The `kotlin-stdlib-common` and `kotlin-stdlib` JAR artifacts are deprecated and replaced by `.klib` files as the standard format for common multiplatform declarations metadata.

Reply:
That sound more scary than it is in reality. Before that we published three JARs under `kotlin-stdlib` coordinates:
- `kotlin-stdlib.jar` - the main artifact for JVM,
- `kotlin-stdlib-all.jar` - with the declarations from all common source sets of the standard library in the modern format,
- `kotlin-stdlib-common.jar` with legacy multiplatform declarations metadata from the commonMain source set only

With this change we stop publishing the last one.

Reply:
Thank you - clarified in the text that it only affects `kotlin-stdlib-common.jar` artifact